Remotion/Data/DomainObjects.UnitTests/Mapping/TypeDefinitionCollectionFactoryTest.cs

```diff
@@ -0,0 +1,103 @@
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+// 
+using System;
+using System.Linq;
+using Moq;
+using NUnit.Framework;
+using Remotion.Data.DomainObjects.Mapping;
+using Remotion.Data.DomainObjects.UnitTests.TestDomain;
+
+namespace Remotion.Data.DomainObjects.UnitTests.Mapping
+{
+  [TestFixture]
+  public class TypeDefinitionCollectionFactoryTest
+  {
+    private TypeDefinitionCollectionFactory _typeDefinitionCollectionFactory;
+    private Mock<IMappingObjectFactory> _mappingObjectFactoryMock;
+    private ClassDefinition _fakeClassDefinition;
+
+    [SetUp]
+    public void SetUp ()
+    {
+      _mappingObjectFactoryMock = new Mock<IMappingObjectFactory>(MockBehavior.Strict);
+      _typeDefinitionCollectionFactory = new TypeDefinitionCollectionFactory(_mappingObjectFactoryMock.Object);
+      _fakeClassDefinition = ClassDefinitionObjectMother.CreateClassDefinitionWithMixins(typeof(Order));
+    }
+
+    [Test]
+    public void CreateTypeDefinitionCollection ()
+    {
+      _mappingObjectFactoryMock
+          .Setup(mock => mock.CreateClassDefinition(typeof(Order), null))
+          .Returns(_fakeClassDefinition)
+          .Verifiable();
+
+      var typeDefinitions = _typeDefinitionCollectionFactory.CreateTypeDefinitionCollection(new[] { typeof(Order) });
+
+      _mappingObjectFactoryMock.Verify();
+
+      Assert.That(typeDefinitions, Is.EqualTo(new[] { _fakeClassDefinition }));
+    }
+
+    [Test]
+    public void CreateTypeDefinitionCollection_NoTypes ()
+    {
+      var typeDefinitions = _typeDefinitionCollectionFactory.CreateTypeDefinitionCollection(new Type[0]);
+
+      Assert.That(typeDefinitions, Is.Empty);
+    }
+
+    [Test]
+    public void CreateTypeDefinitionCollection_DerivedClassAreSet ()
+    {
+      var fakeClassDefinitionCompany = ClassDefinitionObjectMother.CreateClassDefinitionWithMixins(typeof(Company));
+      var fakeClassDefinitionPartner = ClassDefinitionObjectMother.CreateClassDefinition(classType: typeof(Partner), baseClass: fakeClassDefinitionCompany);
+      var fakeClassDefinitionCustomer = ClassDefinitionObjectMother.CreateClassDefinition(classType: typeof(Customer), baseClass: fakeClassDefinitionCompany);
+
+      _mappingObjectFactoryMock
+          .Setup(mock => mock.CreateClassDefinition(typeof(Order), null))
+          .Returns(_fakeClassDefinition)
+          .Verifiable();
+      _mappingObjectFactoryMock
+          .Setup(mock => mock.CreateClassDefinition(typeof(Company), null))
+          .Returns(fakeClassDefinitionCompany)
+          .Verifiable();
+      _mappingObjectFactoryMock
+          .Setup(mock => mock.CreateClassDefinition(typeof(Partner), fakeClassDefinitionCompany))
+          .Returns(fakeClassDefinitionPartner)
+          .Verifiable();
+      _mappingObjectFactoryMock
+          .Setup(mock => mock.CreateClassDefinition(typeof(Customer), fakeClassDefinitionCompany))
+          .Returns(fakeClassDefinitionCustomer)
+          .Verifiable();
```

Comment:
simplification question: just stub these since the result must be verified anyway?